repobee_sanitizer/_sanitize.py

```diff
@@ -26,17 +26,22 @@ def sanitize_text(lines: List[str], strip: bool = False) -> Optional[str]:
 
 def _sanitize(lines: List[str], strip: bool = False) -> Iterable[str]:
     keep = True
-    prefix_length = 0
+    prefix = ""
+    in_replace = False
     for line in lines:
         marker = _syntax.contained_marker(line)
         if marker == Markers.START:
-            prefix = re.match(rf"(.*?){Markers.START.value}", line).group(1)
-            prefix_length = len(prefix) if not strip else 0
+            prefix = (
+                re.match(rf"(.*?){Markers.START.value}", line).group(1).strip()
+            )
```

Comment:
I didn't notice this before, but this should be a function `search_prefix` or something like that as it's duplicated in the syntax check.